src/components/Marketplace/MarketplaceScreen.js

```diff
@@ -70,7 +76,69 @@ export default function MarketplaceScreen() {
   }, []);
 
   // Get total number of results depending on if produce or farm
-  const totalResults = tabValue === 'all' ? produceListings.length : farmListings.length;
+  const totalResults = tabValue === 'all' ? filteredProduce.length : filteredFarms.length;
+
+  // helper fx to check any string for search term
+  const checkFieldForSearchTerm = (field) => {
+    const reg = new RegExp(searchTerms, 'i');
+    return field && reg.test(field);
+  };
+
+  const filterFarmsBySearch = () => {
+    const filtered = farmListings.filter((farm) => checkFieldForSearchTerm(farm.fields['farm name']) === true);
+    setFilteredFarms(filtered);
+  };
+
+  // helper fx to async check produce name
+  const searchProduceName = (produce) => {
```

Comment:
Is there a reason why we have to use an airtable api call for filtering? Can't we do what we did for filterFarmsBySearch and filter our farms with our original state holding our listings?

Reply:
Yeah! The listing objects don't directly reference the names of their produce or farm, the two fields that can be searched for. They only have the id to their produce and farm, so an airtable call needs to be made to fetch the actual names everytime. 

Maybe it could be refactored to hold the names in the original state, that would just take some extra experimenting.